src/chocolatey/infrastructure.app/services/PowershellService.cs

```diff
@@ -503,5 +494,176 @@ private PowerShellExecutionResults run_host(ChocolateyConfiguration config, stri
+                Environment.SetEnvironmentVariable("chocolateyPackageVersion", package.Version.to_string());
+                Environment.SetEnvironmentVariable("packageVersion", package.Version.to_string());
+                Environment.SetEnvironmentVariable("chocolateyPackageFolder", packageDirectory);
+                Environment.SetEnvironmentVariable("packageFolder", packageDirectory);
+                Environment.SetEnvironmentVariable("installArguments", configuration.InstallArguments);
+                Environment.SetEnvironmentVariable("installerArguments", configuration.InstallArguments);
+                Environment.SetEnvironmentVariable("chocolateyInstallArguments", configuration.InstallArguments);
+                Environment.SetEnvironmentVariable("packageParameters", configuration.PackageParameters);
+                Environment.SetEnvironmentVariable("chocolateyPackageParameters", configuration.PackageParameters);
+                if (configuration.ForceX86)
+                {
+                    Environment.SetEnvironmentVariable("chocolateyForceX86", "true");
+                }
+                if (configuration.OverrideArguments)
+                {
+                    Environment.SetEnvironmentVariable("chocolateyInstallOverride", "true");
+                }
+
+                if (configuration.NotSilent)
+                {
+                    Environment.SetEnvironmentVariable("chocolateyInstallOverride", "true");
+                }
+
+                //todo:if (configuration.NoOutput)
+                //{
+                //    Environment.SetEnvironmentVariable("ChocolateyEnvironmentQuiet","true");
+                //}
+
+                foreach (var chocoPowerShellScript in scriptSet)
+                {
+                    this.Log().Debug(ChocolateyLoggers.Important, "Contents of '{0}':".format_with(chocoPowerShellScript.Name));
+                    string chocoPowerShellScriptContents = _fileSystem.read_file(chocoPowerShellScript.Path);
+                    this.Log().Debug(chocoPowerShellScriptContents.escape_curly_braces());
+
+                    bool shouldRun = !configuration.PromptForConfirmation;
+
+                    if (!shouldRun)
+                    {
+                        this.Log().Info(ChocolateyLoggers.Important, () => "The package {0} wants to run '{1}'.".format_with(package.Id, _fileSystem.get_file_name(chocoPowerShellScript.Path)));
+                        this.Log().Info(ChocolateyLoggers.Important, () => "Note: If you don't run this script, the installation will fail.");
+                        this.Log().Info(ChocolateyLoggers.Important, () => @"Note: To confirm automatically next time, use '-y' or consider setting 
+ 'allowGlobalConfirmation'. Run 'choco feature -h' for more details.");
+
+                        var selection = InteractivePrompt.prompt_for_confirmation(@"Do you want to run the script?", new[] { "yes", "no", "print" }, defaultChoice: null, requireAnswer: true);

```

Comment:
Yes. Yes you did. I was _trying_ not to break the existing code while bringing the new stuff online. Clearly (based on the current test failures) I did not succeed. :-( Basically this falls into a "my fault for not doing a PR earlier" and will need to be cleaned up.